### DISCUSSION TITLE
First time user Debian10 AppImage use....

### DISCUSSION THREAD

**Discussion by gebyt-rybrit** [streamlink/streamlink-twitch-gui#779] [General]
Having issue as first time user of this application launching stream.

Using AppImage version on debian 10 -

Trying to launch a stream I get:

Error: Streamlink misconfiguration

Couldn't find executable

Executables were not found

Please check your settings and/or (re)install.

This is a brand new appimage package of streamlink aside from API token I haven't done much else except install MPV and set the player to MPV via dropdown.

my script to launch streamlink appimage is per below ->

export PATH=$PATH:/usr/bin/:/home/debianuser/scripts

OPENSSL_CONF="" QT_SCALE_FACTOR=1 QT_AUTO_SCREEN_SCALE_FACTOR=0 MESA_GLSL_CACHE_DISABLE=true ./streamlink-twitch-gui-v1.11.0-x86_64.AppImage --no-sandbox &

I still get this error in terminal upon launching with script --  (But GUI still launches) [23888:23888:0209/172609.318390:ERROR:component_loader.cc(169)] Failed to parse extension manifest.

and when I click on a Followed stream to try to have it launch, I get in Terminal:

[error][StreamingService]

ProviderError: Couldn't find executable

Executables were not found

In Settings -> Player

I have player preset as MPV

player executable as /usr/bin/mpv

custom parameters --player-operation-mode=pseudo-gui --%U

[reddit link](https://www.reddit.com/r/Streamlink/comments/lghuni/streamlink_for_twitch_on_debian_10_via_appimage/)

**Comment by gebyt-rybrit** [comment]
and yes, the obvious... MPV is installed on my system and working correctly. just now with this appimage, aka per OP not detected....

**Comment by bastimeyer** [comment]
> This is a brand new appimage package of streamlink

There is no Streamlink AppImage. You are confusing this with Streamlink Twitch GUI. Your issue is about Streamlink, not Streamlink Twitch GUI.

> aside from API token I haven't done much else except install MPV

> [error][StreamingService]  
> ProviderError: Couldn't find executable  
> Executables were not found  

**Streamlink Twitch GUI** is a GUI frontend for **Streamlink**. You have not installed Streamlink yet, so the Twitch GUI can't find the streamlink executable to launch the streams.

----

https://github.com/streamlink/streamlink-twitch-gui#description

> Streamlink Twitch GUI  
> A graphical user interface on top of the Streamlink command line interface.

https://github.com/streamlink/streamlink-twitch-gui/wiki/Installation

> Streamlink Twitch GUI requires Streamlink to be installed on the system in order to be able to launch streams.  
> Please follow the install instructions for both applications.

**Reply by gebyt-rybrit** [reply]
oh i see, thank you.  I have installed streamlink, but had to use the backported version instructions since im on debian distribution....

now when it try to launch a twitch stream I get -- 
Error: Invalid Streamlink version - The version 1.0.0 doesn't match the min. requirements (v2.0.0).

Does this mean that (for debian at least) the AppImage is not supported? due to backported version of streamlink?

**Reply by bastimeyer** [reply]
You will need Streamlink >=2.0.0 in order to watch Twitch streams because of changes made by Twitch which broke earlier versions.

If your distro doesn't have the latest Streamlink version in its repos then you can install Streamlink via python-pip as an alternative. See the Streamlink install docs for all the details.

Streamlink 2.0.0 should however be available for `buster-backports`, as mentioned in the Streamlink install docs.
https://packages.debian.org/search?suite=buster-backports&searchon=names&keywords=streamlink

**Reply by gebyt-rybrit** [reply]
im fairly new to this, but I added the buster-backports on my system yet after update repository it only retrieves version 1.0.0:

user@computer:~$ echo "deb http://ftp.us.debian.org/debian buster-backports main" | sudo tee "/etc/apt/sources.list.d/streamlink.list"
deb http://ftp.us.debian.org/debian buster-backports main

user@computer:~$ sudo apt update
Hit:1 http://security.debian.org/debian-security buster/updates InRelease
Hit:2 http://ftp.us.debian.org/debian buster InRelease  
Hit:3 http://ftp.us.debian.org/debian buster-updates InRelease
Get:4 http://ftp.us.debian.org/debian buster-backports InRelease [46.7 kB]
Get:5 http://ftp.us.debian.org/debian buster-backports/main amd64 Packages [433 kB]
Get:6 http://ftp.us.debian.org/debian buster-backports/main Translation-en [369 kB]
Fetched 849 kB in 1s (841 kB/s)         
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.

user@computer:$ sudo apt install streamlink
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
streamlink
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 69.3 kB of archives.
After this operation, 172 kB of additional disk space will be used.
Get:1 http://ftp.us.debian.org/debian buster/main amd64 streamlink all 1.0.0+dfsg-1 [69.3 kB]
Fetched 69.3 kB in 0s (233 kB/s)
Selecting previously unselected package streamlink.
(Reading database ... 254045 files and directories currently installed.)
Preparing to unpack .../streamlink_1.0.0+dfsg-1_all.deb ...
Unpacking streamlink (1.0.0+dfsg-1) ...
Setting up streamlink (1.0.0+dfsg-1) ...
Processing triggers for man-db (2.8.5-2) ...
user@computer:$

will attempt python-pip & follow-up, appreciate the responses!

**Comment by bastimeyer** [comment]
> buster/main amd64 streamlink all 1.0.0+dfsg-1

You are installing from the main debian buster repository...

----

If you take a look at the [Streamlink install docs](https://streamlink.github.io/install.html#linux-and-bsd), it tells you to do this:
```
# If you don't have Debian backports already (see link below):
echo "deb http://deb.debian.org/debian buster-backports main" | sudo tee "/etc/apt/sources.list.d/streamlink.list"

sudo apt update
sudo apt install streamlink
```
And the link below is this:
https://wiki.debian.org/Backports#Using_the_command_line

**On that linked debian wiki page, see the "Installing backports on the command line" section:**

> The backports repository is deactivated by default. So, if you want to install a backported package, you will have to state that explicitly.
>
> For example:
> `# apt -t buster-backports install cockpit`
>
> The -t option here specifies buster-backports as the target release. This would install a newer version of Cockpit and all its reverse dependencies from buster-backports instead of the older one from Debian stable release.

Which means in order to install the streamlink package from buster-backports, you have to run
```
sudo apt -t buster-backports install streamlink
```

**Reply by gebyt-rybrit** [reply]
Thanks! I got it working, I thought by adding back-port repository to the list of available system repositories that it would auto use it to parse packages, I was not considering it required activation. Appreciate the help!
